### DISCUSSION TITLE
Delay postgres triggers until after API response

### DISCUSSION THREAD

**Discussion by kylerummens** [supabase/supabase#14745] [Questions] [answered]
I'm using supabase-js to batch upload data to a table called my_table. I multiple triggers listening on changes to my_table and each of them does quite a bit of processing. This slows down the API response quite a bit, since the triggers are called for every row and must finish before the response is sent back to js. 

I've tried using a deferrable trigger (https://www.postgresql.org/docs/current/sql-createtrigger.html) but with this the trigger is still called during the transaction.

Not only does this slow down the batch processing, but I've even had a number of requests that time out. Is there a way that I can force the API to only perform the insert and not wait for all triggers to complete?

**Comment by steve-chavez** [comment, marked as answer]
> This slows down the API response quite a bit, since the triggers are called for every row

How about using a FOR EACH STATEMENT trigger? With that you can avoid processing per row. Check an example here: https://stackoverflow.com/questions/24193567/for-each-statement-trigger-example

**Reply by kylerummens** [reply]
I had previously ruled out statement-level triggers since you don't have access to the new row, but the link you sent referenced Transition Tables, which I was unaware of. Looks like this will work.

Thanks
